Sources/NIO/ByteBuffer-views.swift

```diff
@@ -187,3 +187,32 @@ extension ByteBuffer {
         self = view._buffer.getSlice(at: view.startIndex, length: view.count)!
     }
 }
+
+extension ByteBufferView: Equatable {
+    /// required by `Equatable`
+    public static func == (lhs: ByteBufferView, rhs: ByteBufferView) -> Bool {
+
+        guard lhs._range.count == rhs._range.count else {
+            return false 
+        }
+
+        let leftBufferSlice = lhs._buffer.getSlice(at: lhs._range.startIndex, length: lhs._range.count)
+        let rightBufferSlice = rhs._buffer.getSlice(at: rhs._range.startIndex, length: rhs._range.count)
```

Comment:
Great, can we force-unwrap these slices? We need to add a comment to explain why we're doing it, which can say:

> A well-formed ByteBufferView can never have a range that is out-of-bounds of the backing ByteBuffer. As a result, these `getSlice` calls can never fail, and we'd like to know it if they do.

Reply:
Thanks for the comment, I've added it as you wrote it as I couldn't come up with anything clearer/briefer than this.